### PR TITLE
gRPC tests for generating and handling commits with different proposals

### PR DESCRIPTION
**Pull request by mulmarta**
* Support more proposal types in the gRPC runner:
  * Group context extensions proposal
  * Resumption PSK proposal with usage application
* Add tests with different proposals committed by reference
  * Create and process a commit with own proposal
  * Create and process a commit with someone else's proposal

### CODE REVIEW

File: interop/test-runner/main.go [hunk 1]
```diff
@@ -682,7 +733,7 @@ func (config *ScriptActorConfig) RunStep(index int, step ScriptStep) error {
 
 		config.StoreMessage(index, "welcome", commitResp.Welcome)
 		config.StoreMessage(index, "commit", commitResp.Commit)
-		if !params.ExternalTree {
+		if params.ExternalTree {
 			config.StoreMessage(index, "ratchet_tree", commitResp.RatchetTree)
 		}
```
Comment (mulmarta) [inline interop/test-runner/main.go:738]: Should this be negated or not?

File: interop/proto/mls_client.proto [hunk 2]
```diff
@@ -184,19 +186,32 @@ message RemoveProposalRequest {
...
   uint32 state_id = 1;
   bytes group_id = 2;
   uint32 cipher_suite = 3; // actually uint16
 }
 
+// rpc GroupContextExtensionsProposal
+message GroupContextExtensionsProposalRequest {
+  uint32 state_id = 1;
+  repeated uint32 extension_types = 2;
+  repeated bytes extension_data = 3;
```
[the remark this answers is not in the document]
Comment (mulmarta) [reply interop/proto/mls_client.proto]: Didn't know one could do that. Thanks!

File: interop/configs/generate_handle_commit.json [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+      {"action": "createKeyPackage", "actor": "alice4"},
+      {"action": "addProposal", "actor": "alice1", "keyPackage": 1},
+      {"action": "addProposal", "actor": "alice1", "keyPackage": 2},
+      {"action": "addProposal", "actor": "alice1", "keyPackage": 3},
+      {"action": "fullCommit", "actor": "alice1", "byReference": [4, 5, 6], "joiners": ["alice2", "alice3", "alice4"]},
+      {"action": "updateProposal", "actor": "alice2"},
+      {"action": "fullCommit", "actor": "alice1", "byReference": [8], "members": ["alice2", "alice3", "alice4"]}
+    ],
+
+    "context_extensions_from_self_and_others": [
```
[the remark this answers is not in the document]
Comment (mulmarta) [reply interop/configs/generate_handle_commit.json]: Removed the text everywhere.

File: interop/configs/commit.json [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    ],
+
+    "remove": [
+      {"action": "createGroup", "actor": "alice"},
+      {"action": "externalJoin", "actor": "alice", "joiner": "bob"},
+      {"action": "createKeyPackage", "actor": "charlie1"},
+      {"action": "createKeyPackage", "actor": "charlie2"},
+      {"action": "addProposal", "actor": "alice", "keyPackage": 2},
+      {"action": "addProposal", "actor": "alice", "keyPackage": 3},
+      {"action": "fullCommit", "actor": "alice", "byReference": [4, 5], "members": ["bob"], "joiners": ["charlie1", "charlie2"]},
```
Comment (bifurcation) [inline interop/configs/commit.json:33]: Is there a reason to do this with `addProposal` and not `externalJoin`?

Comment (bifurcation) [reply interop/configs/commit.json:33]: Also, let's pick a single naming scheme here, either alphabetical (`alice`, `bob`, `charlie`, ...) or numeric (`alice1`, `alice2`, `alice3`, ...)

File: interop/configs/commit.json [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+      {"action": "installExternalPSK", "clients": ["alice", "bob"]},
+
+      {"action": "createKeyPackage", "actor": "charlie3"},
+      {"action": "addProposal", "actor": "alice", "keyPackage": 8},
+      {"action": "removeProposal", "actor": "alice", "removed": "charlie1"},
+      {"action": "groupContextExtensionsProposal", "actor": "alice", "extensions": [
+        {"extension_type": 123, "extension_data": "aGVsbG8="},
+        {"extension_type": 321, "extension_data": "d29ybGQ="}
```
Comment (bifurcation) [inline interop/configs/commit.json:109]: We should normalize on either `camelCase` or `snake_case` for script parameters.  Right now, I think we're largely using `camelCase` for script parameters and `snake_case` for transcript field names.  Maybe we should just use the same for both.

File: interop/configs/commit.json (hunk 4 above)
Comment (mulmarta) [reply interop/configs/commit.json:33]: I forgot to update this one. Should be better now.

File: interop/configs/commit.json (hunk 5 above)
Comment (mulmarta) [reply interop/configs/commit.json:105]: In this case how about a separate PR that makes this consistent?

**Pull request merged**